### DISCUSSION TITLE
Doublet scores

### DISCUSSION THREAD

**Discussion by Kaivalyamolugu** [GreenleafLab/ArchR#618] [Questions / Documentation] [answered]
Hello, 

I am working with 4 samples and the r2 values are 0.89885, 0.88364, 0.85502, 0.67755. Would the data comparisons be biased if I apply the filterdoublets only to the first three samples and not the last one? Or would it be better to not apply filterdoublets at all?

Thanks,
Kaivalya

**Comment by rcorces** [comment, marked as answer]
It is hard to provide specific advice given that this is a somewhat borderline call. You certainly should not attempt doublet filtration on the last sample with r2 = 0.67755. The other three samples are borderline. If this were my experiment, I would try it both ways and see what the consequences of doublet filtration were.

**Reply by Kaivalyamolugu** [reply]
Thank you for the input. Is there a way to filter doublets for 3 of the 4 samples within a single ArchR project?

**Reply by rcorces** [reply]
On second thought, I'm actually not sure if this is possible or how to do it. From what I understand of this, when r2 < 0.9, it marks those samples as `NA` for the doublet scores which prevents them from being filtered. I'm not sure if this can be overridden. Maybe @jgranja24 can confirm?

Here are some relevant issues to check out:
https://github.com/GreenleafLab/ArchR/issues/173
https://github.com/GreenleafLab/ArchR/issues/63
https://github.com/GreenleafLab/ArchR/issues/142

**Reply by Kaivalyamolugu** [reply]
That is helpful. Thank you!

**Comment by Austin-s-h** [comment]
A thought on the same topic-- is it possible to call doublets across an entire (merged) ArchR project? Would this help to increase the difference of pseudo-doublets as long as your samples are continous?

**Reply by rcorces** [reply]
The underlying assumption of doublets is that they come from two cells/nuclei getting incorporated into the same droplet. Since a doublet by this definition can only occur between two cells/nuclei within the same droplet reaction, and we assume that each droplet reaction is represented by a single arrow file, there shouldnt be an improvement in doublet calling by using an entire project. You might improve the R value but I dont think it would improve your ability to actually predict doublets. Maybe I'm not following your train of thought though.

**Reply by Austin-s-h** [reply]
I see what you're saying. Artificially inflating the differences between pseudo-doublets wouldn't actually help the detection of real doublets within the same sample. Thanks!

**Reply by markphillippebworth** [reply]
I have a ground-truth hashed scATAC-seq dataset, and I'd like to see if I get improvement on ArchR doublet call by running it across the entire merged ArchRProject, instead of individual cells. Do you have any tricks on how to do this?

The reason I'm doing this is because I'm seeing strong biases around rarer cell types, and I'm seeing ArchR doublet detection yielding about a 54% sensitivity rate for true doublets in my dataset.

**Reply by rcorces** [reply]
I'm not sure I understand your post. You are effectively trying to take data acquired across multiple individual reactions (multiple arrow files) and call doublets. It is not possible for doublets to exist across arrow files. If you really want to do this, you will need to create a single arrow file. Perhaps you should try changing the parameters for doublet identification instead (for example, `k`)

**Comment by NoemieL** [comment]
Hi, 
I am re-running my samples and get some warnings ( 1)The following arguments are not used: row.names and 2) guides(<scale> = FALSE)` is deprecated. Please use `guides(<scale> = "none")` instead.) and biased clusters. The warnings are also present when running the tutorial.
I wondering how I can manage these warnings. I would also understand the meaning of these biased clusters and if it is a concern for the rest of the analysis. I did not get theses previously. This Thanks

2021-09-15 17:25:10 : Batch Execution w/ safelapply!, 0 mins elapsed.
2021-09-15 17:25:10 : P1498_MM_ATACM (1 of 26) :  Computing Doublet Statistics, 0 mins elapsed.
Warning: The following arguments are not used: row.names
P1498_MM_ATACM (1 of 26) : UMAP Projection R^2 = 0.9913
2021-09-15 17:26:31 : P1389_MM_ATACF (2 of 26) :  Computing Doublet Statistics, 1.353 mins elapsed.
Warning: The following arguments are not used: row.names
Biased Clusters : Cluster20 
P1389_MM_ATACF (2 of 26) : UMAP Projection R^2 = 0.98623
2021-09-15 17:31:59 : P1606_MM_ATACF (3 of 26) :  Computing Doublet Statistics, 6.809 mins elapsed.
Warning: The following arguments are not used: row.names
P1606_MM_ATACF (3 of 26) : UMAP Projection R^2 = 0.99066
2021-09-15 17:34:02 : P1675_MM_ATAC (4 of 26) :  Computing Doublet Statistics, 8.863 mins elapsed.
Warning: The following arguments are not used: row.names
P1675_MM_ATAC (4 of 26) : UMAP Projection R^2 = 0.99404
2021-09-15 17:37:24 : P1395_MM_ATAC (5 of 26) :  Computing Doublet Statistics, 12.238 mins elapsed.
Warning: The following arguments are not used: row.names
Biased Clusters : Cluster12 
P1395_MM_ATAC (5 of 26) : UMAP Projection R^2 = 0.99487
2021-09-15 17:40:29 : P1555_MM_ATAC (6 of 26) :  Computing Doublet Statistics, 15.312 mins elapsed.
Warning: The following arguments are not used: row.names
Warning: The following arguments are not used: row.names
Biased Clusters : Cluster1 
P1555_MM_ATAC (6 of 26) : UMAP Projection R^2 = 0.99588

Warning messages:
1: `guides(<scale> = FALSE)` is deprecated. Please use `guides(<scale> = "none")` instead. 
2: `guides(<scale> = FALSE)` is deprecated. Please use `guides(<scale> = "none")` instead.


With the tutorial
2021-09-15 10:54:15 : Batch Execution w/ safelapply!, 0 mins elapsed.
2021-09-15 10:54:15 : scATAC_BMMC_R1 (1 of 3) :  Computing Doublet Statistics, 0 mins elapsed.
Warning: The following arguments are not used: row.names
scATAC_BMMC_R1 (1 of 3) : UMAP Projection R^2 = 0.98226
2021-09-15 10:55:40 : scATAC_CD34_BMMC_R1 (2 of 3) :  Computing Doublet Statistics, 1.415 mins elapsed.
Warning: The following arguments are not used: row.names
scATAC_CD34_BMMC_R1 (2 of 3) : UMAP Projection R^2 = 0.984
2021-09-15 10:56:48 : scATAC_PBMC_R1 (3 of 3) :  Computing Doublet Statistics, 2.558 mins elapsed.
Warning: The following arguments are not used: row.names
scATAC_PBMC_R1 (3 of 3) : UMAP Projection R^2 = 0.98276
ArchR logging successful to : ArchRLogs/ArchR-addDoubletScores-13334ebfd0630-Date-2021-09-15_Time-10-54-14.log
Warning messages:
1: `guides(<scale> = FALSE)` is deprecated. Please use `guides(<scale> = "none")` instead. 
2: `guides(<scale> = FALSE)` is deprecated. Please use `guides(<scale> = "none")` instead. 
3: `guides(<scale> = FALSE)` is deprecated. Please use `guides(<scale> = "none")` instead.

**Reply by rcorces** [reply]
@NoemieL - duplicate of https://github.com/GreenleafLab/ArchR/issues/1021
Not sure what is going on. Must be something environment specific

**Reply by NoemieL** [reply]
What about the biased clusters?

**Reply by markphillippebworth** [reply]
Biased clusters will pop up when you run doublet detection, but that warning shouldn't create issues downstream.
